### PR TITLE
v1.0.4: adding e2e tests to GBC Deposit UI

### PR DESCRIPTION
**Pull request by Wagalidoom**

### CODE REVIEW

File: hooks/use-autoclaim.ts [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
     async function fetchEvents() {
-      if (contractConfig && account.address && account.address !== "0x0") {
-        const registerEvents = await fetchRegister(contractConfig.addresses.claimRegistry, account.address, contractConfig.claimRegistryStartBlockNumber, client);
-        const unregisterEvents = await fetchUnregister(contractConfig.addresses.claimRegistry, account.address, contractConfig.claimRegistryStartBlockNumber, client);
+      if (contractConfig && address && address !== "0x0") {
+        const registerEvents = await fetchRegister(contractConfig.addresses.claimRegistry, address, contractConfig.claimRegistryStartBlockNumber, client);
+        const unregisterEvents = await fetchUnregister(contractConfig.addresses.claimRegistry, address, contractConfig.claimRegistryStartBlockNumber, client);
 
         if (registerEvents.length == 0 && unregisterEvents.length == 0) {
           setIsRegister(false);
```
Comment (ilge-ustun) [inline hooks/use-autoclaim.ts:25]: Not connected to this PR, but I think it's possible not to use events log (even no subgraph needed) and to read from the `claimRegistry` contract directly as it keeps status in the public `configs` mapping.

Here I've implemented this approach, while playing with envio:
https://github.com/gnosischain/gbc-deposit-ui/blob/3eab96bf1edb979aff9d1f5273d3cf5235b698fb/hooks/use-autoclaim.ts#L21C2-L30C5

File: hooks/use-dappnode-deposit.ts [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            chainId,
+            contractConfig.depositStartBlockNumber
+          );
+
+        const events = await fetchDeposit(
+          contractConfig.addresses.deposit,
+          fromBlock,
+          client
+        );
+
```
Comment (ilge-ustun) [inline hooks/use-dappnode-deposit.ts:116]: Also not connected to this PR, as subgraph wasn't implemented yet, but when indexer will be added, this part will be changed too.

File: hooks/use-dappnode-deposit.ts [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+          return (
+            depositDataJson.pubkey &&
+            depositDataJson.withdrawal_credentials &&
+            depositDataJson.amount &&
+            depositDataJson.signature &&
+            depositDataJson.deposit_message_root &&
+            depositDataJson.deposit_data_root &&
+            depositDataJson.fork_version
+          );
+        };
```
Comment (ilge-ustun) [inline hooks/use-dappnode-deposit.ts:68]: A bit repetitive,  maybe some functionality can be shared with `use-deposit'?

**Review by ilge-ustun — APPROVED** [review]
Impressive work on tests!

For dappnodeDeposit, as improvement, could be nice to refactor the code in the future.
And when we add subgraph - the implementation of dappnode part will slightly change too.

File: hooks/use-autoclaim.ts (hunk 1 above)
Comment (Wagalidoom) [reply hooks/use-autoclaim.ts:25]: Nice improvement i agree! It seems more straightforward than the current implementation 👍

File: hooks/use-dappnode-deposit.ts (hunk 3 above)
Comment (Wagalidoom) [reply hooks/use-dappnode-deposit.ts:68]: I agree too. Also on the UI, i think it makes more sense to move the dappnode deposit component within the basic deposit and have a switcher to on/off dappnode mode.

**Pull request merged**